Sample_v3/Samples/CombineSimpleChat/CombineSimpleChatViewController.swift

```diff
@@ -0,0 +1,437 @@
+    
+    ///
+    /// # subscribeToCombinePublishers
+    ///
+    /// Here we bind `channelControllers` publishers so we can observe the changes.
+    private func subscribeToCombinePublishers() {
+        /// `ChannelController` will not trigger the `channelChangePublisher` on the initial channel set so
+        /// we can` prepend` our `channelChangePublisher` sequence with the initial channel manually.
+        let initialChannel = Just(channelController.channel)
+            .compactMap { $0 }
+            .map { EntityChange<Channel>.update($0) }
+        
+        /// This subscription updates the view controller's `title` and its `navigationItem.prompt` to display the count of channel
+        /// members and the count of online members. When the channel is deleted, this view controller is dismissed.
+        let updatedChannel = channelController
+            .channelChangePublisher
+            /// Update UI for initial channel.
+            .prepend(initialChannel)
+            /// Dismiss VC and break the sequence if channel got deleted.
+            /// Map `EntityChange` to `Channel` and continue executing sequence if it is `update` change.
+            .compactMap { [weak self] change -> Channel? in
+                switch change {
+                case .create:
+                    return nil
+                case let .update(channel):
+                    return channel
+                case .remove:
+                    self?.dismiss(animated: true)
+                    return nil
+                }
+            }
+            .share()
+        
+        updatedChannel
+            .map { $0.extraData.name ?? $0.cid.description }
+            .receive(on: RunLoop.main)
+            .assign(to: \.title, on: self)
+            .store(in: &cancellables)
+        
+        updatedChannel
+            .map { "\($0.members.count) members, \($0.members.filter(\.isOnline).count) online" }
+            .receive(on: RunLoop.main)
+            .assign(to: \.navigationItem.prompt, on: self)
+            .store(in: &cancellables)
+        
+        /// This subscription applies message changes to tableView using custom `Combine` operator.
+        channelController
+            .messagesChangesPublisher
+            /// Apply changes to tableView.
+            .receive(on: RunLoop.main)
+            .sink { [weak self] in self?.tableView.applyListChanges(changes: $0) }
+            .store(in: &cancellables)
+        
+        /// The subscription  below receives a `TypingEvent` and updates the view controller's `navigationItem.prompt` to show that an user is currently typing.
+        channelController
+            .typingEventPublisher
+            /// Map user with the typing event.
+            .map { [weak self] in (self?.channelController.dataStore.user(id: $0.userId), $0) }
+            /// Skip if user was not fetched succesfully from DB.
+            .filter { $0.0 != nil }
```

Comment:
Aren't these two lines equivalent to `compactMap { }` or you split in like this on purpose?

Reply:
I can't see `compactMap` 🧐
`typingEvent` from sequence is never `nil` as well as tuple itself.